tests/MessageTest.cc

```diff
@@ -53,6 +53,8 @@ TEST(MessageTest, testAllocatedContents) {
     Message msg = msgBuilder.build();
     ASSERT_FALSE(strncmp("content", (char*)msg.getData(), msg.getLength()));
     ASSERT_EQ(content, (char*)msg.getData());
+
+    ASSERT_THROW(msg.releaseData(), std::runtime_error);
```

Comment:
Please add a test case for releasing the data successfully and verify the data is actually swapped.